server/src/unifyfs_request_manager.c

```diff
@@ -1022,6 +1022,142 @@ int create_gfid_chunk_reads(reqmgr_thrd_t* thrd_ctrl,
     return rc;
 }
 
+/* return number of slice ranges needed to cover range */
+static size_t num_slices(size_t offset, size_t length)
+{
+    size_t start = offset / max_recs_per_slice;
+    size_t end   = (offset + length - 1) / max_recs_per_slice;
+    size_t count = end - start + 1;
+    return count;
+}
+
+static int split_request(
```

Comment:
Please add a comment on what this function does, it's arguments, and its return value.